controlplane/kubeadm/controllers/controller.go

```diff
@@ -147,6 +148,12 @@ func (r *KubeadmControlPlaneReconciler) Reconcile(req ctrl.Request) (res ctrl.Re
 			reterr = kerrors.NewAggregate([]error{reterr, err})
 		}
 
+		// Make KCP to requeue in case status is not ready, so we can check for node status without waiting for a re-sync.
+		// TODO: remove this as soon as we have a proper remote cluster cache in cache.
+		if !kcp.Status.Ready && !res.Requeue && kcp.ObjectMeta.DeletionTimestamp.IsZero() {
+			res = ctrl.Result{Requeue: true, RequeueAfter: 10 * time.Second}
```

Comment:
```suggestion
			res = ctrl.Result{RequeueAfter: 20 * time.Second}
```

10s seems a little bit much

Reply:
This should probably move after the patch, also check that `reterr` isn't nil

Reply:
fixed!